output/http/outputhttp.go

```diff
@@ -98,29 +133,121 @@ func (t *OutputConfig) Output(ctx context.Context, event logevent.LogEvent) (err
 
 	resp, err := t.httpClient.Do(req)
 	if err != nil {
+		t.failedDelivery(ctx, event)
 		return err
 	}
 	defer resp.Body.Close()
 
 	_, err = ioutil.ReadAll(resp.Body)
 	if err != nil {
+		t.failedDelivery(ctx, event)
 		return err
 	}
-	if !t.checkIntInList(resp.StatusCode) {
-		err = ErrEndpointDown1.New(nil, url)
-		goglog.Logger.Errorf("output http: %v, status=%v", err, resp.StatusCode)
-		return err
+	if t.checkIntInDiscardedList(resp.StatusCode) {
+		return ErrPermanentError.New(nil, url, resp.StatusCode)
+	}
+	if !t.checkIntInAcceptedList(resp.StatusCode) {
+		t.failedDelivery(ctx, event)
+		return ErrSoftError.New(nil, url, resp.StatusCode)
+	}
+	// the event was sent correctly, we now have to resume inputs if we have requested a pause.
+	if atomic.CompareAndSwapUint32(&t.isInPause, statusPaused, statusDelivering) {
+		goglog.Logger.Debug("outputhttp requesting resume")
+		err := t.control.RequestResume(ctx)
+		if err != nil {
+			goglog.Logger.Error("outputhttp", err.Error())
+		}
 	}
-
 	return nil
 }
 
-// checkIntInList checks if code is in configured list of accepted status codes
-func (t *OutputConfig) checkIntInList(code int) bool {
+// checkIntInAcceptedList checks if code is in configured list of accepted status codes
+func (t *OutputConfig) checkIntInAcceptedList(code int) bool {
 	for _, v := range t.AcceptedHttpResult {
 		if v == code {
 			return true
 		}
 	}
 	return false
 }
+
+// checkIntInAcceptedList checks if code is in configured list of status codes where we should discard the message
```

Comment:
`checkIntInAcceptedList` -> `checkIntInDiscardedList`

Reply:
The price of copy & paste. Fixed.